### PR TITLE
Replace chrono with time

### PR DESCRIPTION
**Pull request by ShellWowza**
Resolves `cargo audit` flagging this crate's dependencies as having vulnerabilities.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 [package]
 name = "slog-bunyan"
 version = "2.3.0"
+edition = "2018"
```
Comment (Techcable) [inline Cargo.toml]: Why did you update to 2018 here?

~~Not criticizing, just curious.
In fact, I'll go separate this out into a separate commit right now~~

**EDIT:** I separated this out into PR #12

File: lib.rs [hunk 2]
```diff
@@ -105,19 +99,25 @@ pub fn with_name<W>(name: &'static str, io: W) -> slog_json::JsonBuilder<W>
 where
     W: io::Write,
 {
-    new_with_ts_fn(name, io, |_: &Record| chrono::Local::now().to_rfc3339())
+    new_with_ts_fn(name, io, |_: &Record| {
+        time::OffsetDateTime::now_local()
+            .unwrap_or_else(|_| time::OffsetDateTime::now_utc())
+            .format(&time::format_description::well_known::Rfc3339)
+            .ok()
```
Comment (Techcable) [inline lib.rs:106]: Why does this implicitly suppress errors here?

Are there alternatives?

File: lib.rs [hunk 3]
```diff
@@ -163,7 +165,7 @@ mod test {
                 + "\"v\":0,"
                 + "\"name\":\"slog-rs\","
                 + "\"level\":30,"
-                + "\"time\":\"2014-07-08T09:10:11+00:00\","
+                + "\"time\":\"2014-07-08T09:10:11Z\","
```
Comment (Techcable) [inline lib.rs:168]: Hi! Why are we changing the time format here?

Is it just for convenience, or is there some other reason?

**Comment by Techcable** [comment]
Hi. Thanks so much for this PR! This definitely needs to be done.

I've separated out the upgrade to Rust 2018 into a separate PR (#12)

One (small) question, why did you change the format for the times? Was there a specific reason or is it just convenience?

File: lib.rs (hunk 2 above)
Comment (Techcable) [reply lib.rs:106]: So looking at the time documentation, formatting errors can only happen if:
1. There is [`InsufficientTypeInformation`](https://docs.rs/time/latest/time/error/enum.Format.html#variant.InsufficientTypeInformation)
2. There is [`InvalidComponent`](https://docs.rs/time/latest/time/error/enum.Format.html#variant.InvalidComponent) 
3. There is an internal `std::io::Error`

None of these are really concerns for us 😉

**Review by Techcable — CHANGES_REQUESTED** [review]
Great job. I separated out the Rust 2018 changes into #11 

Only thing I'm concerned about is changing the time format.

File: lib.rs (hunk 3 above)
Comment (ShellWowza) [reply lib.rs:168]: Hey, sorry, just got back to work from a long weekend. The reason here is simply a minor difference between chrono and time's implementation of RFC3339; Z and +00:00 are both valid timezone specifiers meaning the same thing per the standard. I could implement a custom format description which forces +00:00 instead; would this be prefered?

Comment (ShellWowza) [reply lib.rs:168]: References for this being correct:

[bunyan docs](https://www.npmjs.com/package/bunyan):

> `time`: Required. String. Added by Bunyan. Can be overridden. The date and time of the event in ISO 8601 Extended Format format and in UTC, as from [Date.toISOString()](https://developer.mozilla.org/en/JavaScript/Reference/Global_Objects/Date/toISOString).

[Date.prototype.toISOString()](https://developer.mozilla.org/en/JavaScript/Reference/Global_Objects/Date/toISOString):

> The toISOString() method returns a string in simplified extended ISO format ([ISO 8601](https://en.wikipedia.org/wiki/ISO_8601)), which is always 24 or 27 characters long (YYYY-MM-DDTHH:mm:ss.sssZ or ±YYYYYY-MM-DDTHH:mm:ss.sssZ, respectively). The timezone is always zero UTC offset, as denoted by the suffix "Z".

Comment (Techcable) [reply lib.rs:168]: Wow. I really appreciate the investigation here. Looks like you resolved a bug instead of introducing it :)

**Review by Techcable — APPROVED** [review]
Thanks so much for this. I really appreciate this PR :)

**Comment by Techcable** [comment]
One thing to note is that this increases the Minimum Supported Rust Version from `1.38` to `1.53`.

I am 100% cool with this. However, this means next release we will need to:
1. Indicate this in the Changelog
2. bump the minor version from `2.3` -> `2.4`.

Additionally, I am also planning to switch from travis (broken) to Github Actions next release. That means we will never bump the MSRV by accident again 🎉 😄

**Pull request merged**